src/components/structures/RoomStatusBar.tsx

```diff
@@ -142,6 +153,15 @@ export default class RoomStatusBar extends React.PureComponent<IProps, IState> {
         dis.fire(Action.FocusSendMessageComposer);
     };
 
+    private onRefreshTimelineClick = (): void => {
+        // Empty out the current timeline and re-request it
+        this.props.room.refreshLiveTimeline();
+
+        this.setState({
+            timelineNeedsRefresh: false,
+        });
+    };
```

Comment:
I think the behavior of all of this would be best served by an end-to-end test. It looks like we just [added Cypress recently](https://github.com/matrix-org/matrix-react-sdk/pull/8295) but it's missing all of the utility necessary to complete something like this.

 - Needs a command to create a user behind the scenes and visit Element as the user already signed in
 - Needs commands to setup rooms and send events
 - Needs a way to enable `experimental_features` in the `homeserver.yaml` template
 - Needs a way to add an application service registration in the Synapse instance. The MSC2716 `/batch_send` endpoint is only accessible from a AS token. No extra AS server needed, just the AS token configured.
 - Needs a way to interact with the homserver directly from the application service token to call `/batch_send` (probably via `matrix-js-sdk`)
 
I can't find an overall issue describing the need/want to use Cypress so it's unclear how much we want to move forward with it. I've had [many troubles using Cypress](https://gitlab.com/gitlab-org/quality/team-tasks/-/issues/167#note_293721666) with Gitter.

It seems like we have [other e2e tests using Puppeteer](https://github.com/matrix-org/matrix-react-sdk/tree/27118a97993ea267eb6db398f7098ce61c5ff4b0/test/end-to-end-tests) but I'm guessing we want to move all of these to Cypress. Better place I should be adding some e2e tests or approaching this?

Reply:
> I can't find an overall issue describing the need/want to use Cypress so it's unclear how much we want to move forward with it. I've had many troubles using Cypress with Gitter.

> It seems like we have [other e2e tests using Puppeteer](https://github.com/matrix-org/matrix-react-sdk/tree/27118a97993ea267eb6db398f7098ce61c5ff4b0/test/end-to-end-tests) but I'm guessing we want to move all of these to Cypress. Better place I should be adding some e2e tests or approaching this?

Cypress will be replacing Puppeteer for all E2E test usages in EW iteratively.

Reply:
Can I get some assistance from the Element Web team for at least creating up some of the basic commands needed in Cypress? I don't want to tread on the taste-making here since it's so early and it sounds like you have a plan to move things over which will use these patterns.

> - Needs a command to create a user behind the scenes and visit Element as the user already signed in
> - Needs commands to setup rooms and send events
> [...]
> - Needs a way to interact with the homserver directly from the application service token to call `/batch_send` (probably via `matrix-js-sdk`)

I can add the extras needed for my specific case here.

Reply:
> Can I get some assistance from the Element Web team for at least creating up some of the basic commands needed in Cypress?

Yes but its unlikely to get any attention anytime soon due to other projects e.g Threads being much higher priority

Reply:
It seems like threads or any other feature being worked on could use some e2e tests so they don't seem as mutually exclusive. Even the Cypress [performance tests mentioned in last weeks Element Chapter sync](https://docs.google.com/document/d/11-rd31tZxCvtp-8hfGj3sKlAE1m1r9L1n8aklR5ZLqU/edit#bookmark=id.qal90jyghne5) would probably use these commands.

Reply:
Threads already had e2ee tests written for it - in the Puppeteer suite.

Reply:
> Even the Cypress [performance tests mentioned in last weeks Element Chapter sync](https://docs.google.com/document/d/11-rd31tZxCvtp-8hfGj3sKlAE1m1r9L1n8aklR5ZLqU/edit#bookmark=id.qal90jyghne5) would probably use these commands.

Yup - my comment predated that sync by a day